pyomo/contrib/incidence_analysis/util.py

```diff
@@ -0,0 +1,98 @@
+#  ___________________________________________________________________________
+#
+#  Pyomo: Python Optimization Modeling Objects
+#  Copyright 2017 National Technology and Engineering Solutions of Sandia, LLC
+#  Under the terms of Contract DE-NA0003525 with National Technology and 
+#  Engineering Solutions of Sandia, LLC, the U.S. Government retains certain 
+#  rights in this software.
+#  This software is distributed under the 3-clause BSD License.
+#  ___________________________________________________________________________
+
+from pyomo.core.base.var import Var
+from pyomo.core.base.constraint import Constraint
+from pyomo.util.calc_var_value import calculate_variable_from_constraint
+from pyomo.util.subsystems import (
+        create_subsystem_block,
+        TemporarySubsystemManager,
+        generate_subsystem_blocks,
+        )
+from pyomo.contrib.incidence_analysis.interface import IncidenceGraphInterface
+
+
+def generate_strongly_connected_components(
+        block,
+        include_fixed=False,
+        fix_inputs=True,
+        ):
+    """ Performs a block triangularization of the variable-constraint
+    incidence matrix of the provided block, and yields a block that
+    contains the variables and constraints of each diagonal block
+    (strongly connected component).
+    """
```

Comment:
Please document the `include_fixed` and `fix_inputs` parameters

Reply:
Done